server/src/main/java/org/apache/cassandra/sidecar/config/yaml/InstanceConfigurationImpl.java

```diff
@@ -92,9 +112,13 @@ public InstanceConfigurationImpl(int id,
         this.id = id;
         this.host = host;
         this.port = port;
-        this.dataDirs = Collections.unmodifiableList(dataDirs);
+        this.dataDirs = dataDirs == null ? null : Collections.unmodifiableList(dataDirs);
```

Comment:
when would data dirs be null (or empty)? Maybe this needs to be validated instead of accepting null

Reply:
> when would data dirs be null (or empty)? Maybe this needs to be validated instead of accepting null

It is set to null if the data_dirs is not configured in sidecar.yaml. I was not sure why it was allowed to be null earlier. So did not change the behavior. I can make changes now though.